readthedocs/proxito/tests/test_redirects.py

```diff
@@ -34,6 +34,63 @@ def test_root_redirect_with_query_params(self):
             'https://project.dev.readthedocs.io/en/latest/?foo=bar'
         )
 
+    def test_canonicalize_https_redirect(self):
+        self.domain.canonical = True
+        self.domain.save()
+
+        r = self.client.get('/', HTTP_HOST=self.domain.domain)
+        self.assertEqual(r.status_code, 302)
+        self.assertEqual(
+            r['Location'], f'https://{self.domain.domain}/en/latest/',
```

Comment:
We are doing only 1 redirect now, instead of 2. I think we were doing,

1. http -> https
2. https -> /en/latest/

Now we are doing http -> https /en/latest/ in one step. Is this correct?

Example,

```
$ curl --silent -IL http://docs.readthedocs.io/ | grep Location
Location: https://docs.readthedocs.io/
Location: https://docs.readthedocs.io/en/stable/
```

Reply:
Ideally there are two redirects. I actually think that for `*.readthedocs.io` there will be two since the http -> https redirect will be handled by nginx before it hits proxito. For custom domains that need to both change domain and scheme, this change does use a single redirect.